src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,261 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+    // user created txid1, bumped the fee which resulted in txid2, and now wishes to bump the fee
+    // again, the user must call bumpfee on txid2.
+    const CWalletTx *conflictTx;
+    BOOST_FOREACH(const uint256& conflict, wtx.GetConflicts()) {
+        conflictTx = pwalletMain->GetWalletTx(conflict);
+        LogPrint("rpc", "bumpfee wallet conflict, conflicted tx has time: %d\n", conflictTx->GetTxTime());
+        if (conflictTx->GetTxTime() > wtx.GetTxTime()) {
+            strError = strprintf("Transaction conflicts with later wallet tx: %s", conflict.GetHex());
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strError);
+        }
+    }
+
+    // bounds-check the output to decrement
+    int nOutput = request.params[1].get_int();
+    if (nOutput < 0 || (unsigned int) nOutput > wtx.vout.size())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Output out of bounds");
+
+    // optional parameters
+    int newConfirmTarget = nTxConfirmTarget;
+    CAmount totalFee = 0;
+    CAmount bumpfeeMaxTxFee = 0.01 * COIN;
+    if (request.params.size() > 2) {
+        RPCTypeCheck(request.params, boost::assign::list_of(UniValue::VSTR)(UniValue::VNUM)(UniValue::VOBJ));
+        UniValue options = request.params[2];
+        if (options.size() > 3)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Too many optional parameters");
+        RPCTypeCheckObj(options,
+            {
+                {"confTarget", UniValueType(UniValue::VNUM)},
+                {"totalFee", UniValueType(UniValue::VNUM)},
+                {"maxFee", UniValueType(UniValue::VNUM)},
+            },
+            true, true);
+
+        if (options.exists("confTarget")) {
+            newConfirmTarget = options["confTarget"].get_int();
+            if (newConfirmTarget <= 0) // upper-bound will be checked by estimatefee/smartfee
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid confTarget (cannot be <= 0)");
+        }
+        if (options.exists("maxFee")) {
+            CAmount proposedMaxFee = options["maxFee"].get_int();
+            if (proposedMaxFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid maxFee (cannot be <= 0)");
+            else if (proposedMaxFee > maxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid maxFee (cannot be higher than the global maxTxFee)");
+            bumpfeeMaxTxFee = proposedMaxFee;
+        }
+        if (options.exists("totalFee")) {
+            totalFee = options["totalFee"].get_int();
+            if (totalFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be <= 0)");
+            else if (totalFee > bumpfeeMaxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be higher than bumpfee's maxFee)");
+        }
+    }
+
+    // calculate the old fee and fee-rate
+    CAmount nDebit = wtx.GetDebit(ISMINE_SPENDABLE);
```

Comment:
We need to be sure we are calculating the correct fee here.  Either using new `IsAllFromMe` or getting the fee from the mempool.